ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/navigation/CustomerCenterNavigationState.kt

```diff
@@ -0,0 +1,76 @@
+package com.revenuecat.purchases.ui.revenuecatui.customercenter.navigation
+
+import androidx.compose.runtime.Immutable
+import com.revenuecat.purchases.ui.revenuecatui.helpers.Logger
+import java.util.ArrayDeque
+import java.util.Deque
+
+@Immutable
+internal data class CustomerCenterNavigationState(
+    private val mainTitle: String?,
+    val backStack: Deque<CustomerCenterDestination> = ArrayDeque<CustomerCenterDestination>().apply {
+        // Setting this to null allows the Main destination to be displayed without a title until we have a top bar
+        push(CustomerCenterDestination.Main(null))
+    },
+) {
+    val currentDestination: CustomerCenterDestination
+        get() = backStack.peek() ?: CustomerCenterDestination.Main(title = null)
```

Comment:
nit: what about `backDestination`? I know I said `destination` before hehe, but this is really the top of the backStack, so unless I am missing something it's "where to go if back is pressed", right?

Reply:
FWIW, `back stack` is a pretty common Android term, and it's what this is... So I like it as it is 😇 . Also, I think this includes the "current" destination in the stack right? So it's the stack that gets popped every time the back button is pressed I guess 😅